matrix_sdk/src/room/common.rs

```diff
@@ -0,0 +1,104 @@
+use matrix_sdk_common::api::r0::{
+    membership::{forget_room, get_member_events, join_room_by_id, leave_room},
+    message::get_message_events,
+};
+use std::ops::Deref;
+
+use crate::{Client, Result, Room, RoomMember};
+
+/// A struct containing methodes that are common for Joined, Invited and Left Rooms
+#[derive(Debug, Clone)]
+pub struct Common {
+    inner: Room,
+    pub(crate) client: Client,
+}
+
+impl Deref for Common {
+    type Target = Room;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl Common {
+    /// Create a new `room::Common`
+    ///
+    /// # Arguments
+    /// * `client` - The client used to make requests.
+    ///
+    /// * `room` - The underlaying room.
+    pub fn new(client: Client, room: Room) -> Self {
+        Self {
+            inner: room,
+            client,
+        }
+    }
+
+    /// Forget this room.
+    ///
+    /// This communicates to the homeserver that it should forget the room.
+    pub async fn forget(&self) -> Result<()> {
```

Comment:
Only left rooms can be forgotten as far as I know.

Reply:
From the docs:
> If the user is currently joined to the room, they must leave the room before calling this API.

You are right, i didn't see the above statement before. Gonna move it.